imagekit-sdk/src/test/java/io/imagekit/sdk/ImageKitTest.java

```diff
@@ -2,629 +2,1890 @@
+		options.put("src", "https://ik.imagekit.io/your_imagekit_id/endpoint/default-image.jpg?q=abc&&");
+		options.put("queryParameters", queryParams);
+		options.put("transformation", transformation);
+
+		String url = SUT.getUrl(options);
+		assertThat("https://ik.imagekit.io/your_imagekit_id/endpoint/default-image.jpg?ik-sdk-version="
+				+ Version.VERSION_CODE + "&q=abc&v=123&z=234&tr=w-400,h-600", is(url));
+	}
+
+	@Test
+	public void getUrl_with_double_question_mark_check() {
+		Map<String, String> queryParams = new HashMap<>();
+		queryParams.put("v", "123");
+		queryParams.put("z", "234");
+
+		List<Map<String, String>> transformation = new ArrayList<Map<String, String>>();
+		Map<String, String> scale = new HashMap<>();
+		scale.put("height", "600");
+		scale.put("width", "400");
+		transformation.add(scale);
+
+		Map<String, Object> options = new HashMap<>();
+		options.put("src", "https://ik.imagekit.io/your_imagekit_id/endpoint/default-image.jpg??");
+		options.put("queryParameters", queryParams);
+		options.put("transformation", transformation);
+
+		String url = SUT.getUrl(options);
+		assertThat("https://ik.imagekit.io/your_imagekit_id/endpoint/default-image.jpg?ik-sdk-version="
+				+ Version.VERSION_CODE + "&v=123&z=234&tr=w-400,h-600", is(url));
+	}
+
+	@Test
+	public void getUrl_with_src() {
+		List<Map<String, String>> transformation = new ArrayList<Map<String, String>>();
+		Map<String, String> scale = new HashMap<>();
+		scale.put("height", "600");
+		scale.put("width", "400");
+		transformation.add(scale);
+
+		Map<String, Object> options = new HashMap<>();
+		options.put("src", "https://ik.imagekit.io/your_imagekit_id/endpoint/default-image.jpg");
+		options.put("transformation", transformation);
+
+		String url = SUT.getUrl(options);
+		assertThat("https://ik.imagekit.io/your_imagekit_id/endpoint/default-image.jpg?ik-sdk-version="
+				+ Version.VERSION_CODE + "&tr=w-400,h-600", is(url));
+	}
+
+	@Test
+	public void getUrl_src_with_query_params() {
+		Map<String, String> queryParams = new HashMap<>();
+		queryParams.put("v", "123");
+		queryParams.put("z", "234");
+
+		List<Map<String, String>> transformation = new ArrayList<Map<String, String>>();
+		Map<String, String> scale = new HashMap<>();
+		scale.put("height", "600");
+		scale.put("width", "400");
+		transformation.add(scale);
+
+		Map<String, Object> options = new HashMap<>();
+		options.put("src", "https://ik.imagekit.io/your_imagekit_id/endpoint/default-image.jpg?srcParam=srcParamValue");
+		options.put("queryParameters", queryParams);
+		options.put("transformation", transformation);
+
+		String url = SUT.getUrl(options);
+		assertThat("https://ik.imagekit.io/your_imagekit_id/endpoint/default-image.jpg?ik-sdk-version="
+				+ Version.VERSION_CODE + "&srcParam=srcParamValue&v=123&z=234&tr=w-400,h-600", is(url));
+	}
+
+	@Test
+	public void getUrl_src_with_query_params_but_transformationPosition_is_path() {
+		Map<String, String> queryParam = new HashMap<>();
```

Comment:
Include a gigantic test case that coveres all the parameter including `raw` like this https://github.com/imagekit-developer/imagekit-ruby/blob/master/test/imagekit/url_test.rb#L142

Notice how value of `overlay_image` and `default_image` has been changed such that `/` is replaced with `@@`.
Docs - https://docs.imagekit.io/features/image-transformations/overlay#using-overlay-image-that-are-stored-in-a-nested-folder